src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplate.java

```diff
@@ -135,41 +135,49 @@
 	private static final String FIELD_SCORE = "_score";
 
 	private Client client;
-	private ElasticsearchConverter elasticsearchConverter;
+	private ElasticsearchEntityConverter elasticsearchEntityConverter;
 	private ResultsMapper resultsMapper;
 	private String searchTimeout;
 
 	public ElasticsearchTemplate(Client client) {
-		this(client, new MappingElasticsearchConverter(new SimpleElasticsearchMappingContext()));
+		MappingElasticsearchConverter mappingElasticsearchConverter = createElasticsearchConverter();
+		initialize(client, mappingElasticsearchConverter,
+				new DefaultResultMapper(mappingElasticsearchConverter.getMappingContext()));
 	}
 
-	public ElasticsearchTemplate(Client client, EntityMapper entityMapper) {
-		this(client, new MappingElasticsearchConverter(new SimpleElasticsearchMappingContext()), entityMapper);
-	}
-
-	public ElasticsearchTemplate(Client client, ElasticsearchConverter elasticsearchConverter,
+	public ElasticsearchTemplate(Client client, ElasticsearchEntityConverter elasticsearchEntityConverter,
```

Comment:
Let's also get rid of `EntityMapper`. That is something to be moved into `MappingElasticsearchConverter`.

Reply:
This will be done after the `ResultMapper` has been merged into `MappingElasticsearchConverter` because `EntityMapper` is used there.